pkg/controller/instance/instance_controller.go

```diff
@@ -536,5 +576,8 @@ func scheduledPlan(i *kudoapi.Instance, ov *kudoapi.OperatorVersion) (string, ty
 		i.Spec.PlanExecution.Status = kudoapi.ExecutionNeverRun
 	}
 
+	if i.Spec.PlanExecution.PlanName == kudoapi.DeployPlanName || i.Spec.PlanExecution.PlanName == kudoapi.UpgradePlanName || i.Spec.PlanExecution.PlanName == kudoapi.UpdatePlanName {
+		i.SetReadinessUnknown()
```

Comment:
Again, this "simple getter" (with an exception) shouldn't update Instance status as a side effect. Let's move this someplace else.

Reply:
how is this a getter if this method modifies Spec? :D

Reply:
I meant in [this comment](https://github.com/kudobuilder/kudo/pull/1706#discussion_r501117481) that this is an ugly but necessary side-effect that is at least about the plan name. Instance status modification is an even bigger one.